From f68c1a74da7279f32550fe0b1eeb334079bf2687 Mon Sep 17 00:00:00 2001
From: Daan Leijen <daan@microsoft.com>
Date: Fri, 29 Jan 2021 14:33:49 -0800
Subject: [PATCH] fix assertion comparison (#353)

---
 src/heap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/heap.c b/src/heap.c
index db1b773..54562d1 100644
--- a/src/heap.c
+++ b/src/heap.c
@@ -359,7 +359,7 @@ static void mi_heap_absorb(mi_heap_t* heap, mi_heap_t* from) {
   // turns out to be ok as `_mi_heap_delayed_free` only visits the list and calls a 
   // the regular `_mi_free_delayed_block` which is safe.
   _mi_heap_delayed_free(from);  
-  mi_assert_internal(from->thread_delayed_free == NULL);
+  mi_assert_internal(mi_atomic_load_ptr_relaxed(mi_block_t,&from->thread_delayed_free) == NULL);
 
   // and reset the `from` heap
   mi_heap_reset_pages(from);  
